tls/s2n_handshake_io.c

```diff
@@ -1176,6 +1186,11 @@ static int s2n_handshake_write_io(struct s2n_connection *conn)
     uint8_t record_type = EXPECTED_RECORD_TYPE(conn);
     s2n_blocked_status blocked = S2N_NOT_BLOCKED;
 
+    /* Ensure the state machine referenced is consistent throughout the handshake */
+    if (conn->handshake.state_machine_fixed) {
+        POSIX_ENSURE(IS_TLS13_HANDSHAKE(conn) == conn->handshake.tls13_state_machine, S2N_ERR_STATE_MACHINE_SWITCH);
+    }
```

Comment:
Why do we check that the state machine we choose matches what we expect, rather than choosing the state machine we expect? Like why not choose the state machine based off of the new fields rather than off of actual_protocol_version?

Is it because that would require SAW updates?